internal/provider/resource_gitlab_service_emails_on_push_test.go

```diff
@@ -0,0 +1,146 @@
+		ProviderFactories: providerFactories,
+		CheckDestroy:      testAccCheckGitlabServiceEmailsOnPushDestroy,
+		Steps: []resource.TestStep{
+			// Create an Emails on Push service
+			{
+				Config: testAccGitlabServiceEmailsOnPushConfig(testProject.ID, recipients1),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckGitlabServiceEmailsOnPushExists(emailsOnPushResourceName, &emailsOnPushService),
+					resource.TestCheckResourceAttr(emailsOnPushResourceName, "recipients", recipients1),
+					resource.TestCheckResourceAttr(emailsOnPushResourceName, "active", "true"),
+					resource.TestCheckResourceAttrWith(emailsOnPushResourceName, "created_at", func(value string) error {
+						expectedValue := emailsOnPushService.CreatedAt.Format(time.RFC3339)
+						if value != expectedValue {
+							return fmt.Errorf("should be equal to %s", expectedValue)
+						}
+						return nil
+					}),
+				),
+			},
+			// Verify import
+			{
+				ResourceName:      "gitlab_service_emails_on_push.this",
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+			// Update the Emails on Push service
+			{
+				Config: testAccGitlabServiceEmailsOnPushConfig(testProject.ID, recipients2),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckGitlabServiceEmailsOnPushExists(emailsOnPushResourceName, &emailsOnPushService),
+					resource.TestCheckResourceAttr(emailsOnPushResourceName, "recipients", recipients2),
+					resource.TestCheckResourceAttrWith(emailsOnPushResourceName, "created_at", func(value string) error {
+						expectedValue := emailsOnPushService.CreatedAt.Format(time.RFC3339)
+						if value != expectedValue {
+							return fmt.Errorf("should be equal to %s", expectedValue)
+						}
+						return nil
+					}),
+					resource.TestCheckResourceAttrWith(emailsOnPushResourceName, "updated_at", func(value string) error {
+						expectedValue := emailsOnPushService.UpdatedAt.Format(time.RFC3339)
+						if value != expectedValue {
+							return fmt.Errorf("should be equal to %s", expectedValue)
+						}
+						return nil
+					}),
+				),
+			},
+			// Verify import
+			{
+				ResourceName:      "gitlab_service_emails_on_push.this",
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+			// Update the Emails on Push service to get back to previous settings
+			{
+				Config: testAccGitlabServiceEmailsOnPushConfig(testProject.ID, recipients1),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckGitlabServiceEmailsOnPushExists(emailsOnPushResourceName, &emailsOnPushService),
+					resource.TestCheckResourceAttr(emailsOnPushResourceName, "recipients", recipients1),
+				),
+			},
+			// Verify import
+			{
+				ResourceName:      "gitlab_service_emails_on_push.this",
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
+func testAccCheckGitlabServiceEmailsOnPushExists(resourceIdentifier string, service *gitlab.EmailsOnPushService) resource.TestCheckFunc {
```

Comment:
I don't think this adds any value. The import test step will actually ensure it works properly. 

See here https://github.com/gitlabhq/terraform-provider-gitlab/issues/1008#:~:text=and%20data%20sources.-,Use%20ImportStateVerify%20TestSteps%20as%20a%20replacement%20for%20writing%20custom%20%22CheckExists%22%20and%20%22CheckAttributes%22%20check%20funcs.,-Undesirable%3A

WDYT?

Reply:
yes, looking at it, it seems to be already fully covered by ImportVerify